core/lib/dal/src/consensus_dal.rs

```diff
@@ -402,6 +440,48 @@ impl ConsensusDal<'_, '_> {
         Ok(())
     }
 
+    pub async fn insert_batch_committees(
+        &mut self,
+        number: BatchNumber,
+        validator_committee: ValidatorCommittee,
+        attester_committee: AttesterCommittee,
+    ) -> anyhow::Result<()> {
```

Comment:
if we are going to query the last batch with committees/certificate, then keeping them in the same table makes little sense, because we will need extra indices for efficient lookups. Do we really want to go this way?

Reply:
Querying the last batch with committees is a one-time read on server start, so I didn’t thought it’s a big deal. I don’t see such an index for the query of the last batch with certificate (unless it’s applied internally via the toolchain).

In the overall, I’m not against separating the tables, and originally suggested that. IIRC @RomanBrodetski thought it’s an overkill, and that having all info in respect to a given batch in one place is useful.

Reply:
there is no index, because the the table serves as index for itself when it has just 1 non null column. With 2 columns, it won't be the case.

Reply:
>  because we will need extra indices for efficient lookups.

why exactly? `select ... order by ID desc where certificate is not null` will only scan the rows for which there is committee but not certificate. Given that it happens on startup, it shouldn't be a problem - am I missing smt?

Reply:
> there is no index, because the the table serves as index for itself when it has just 1 non null column. With 2 columns, it won't be the case.

Right, I overlooked that in that case there's no need for `WHERE` clause.

> why exactly? `select ... order by ID desc where certificate is not null` will only scan the rows for which there is committee but not certificate. Given that it happens on startup, it shouldn't be a problem - am I missing smt?

If column X is NOT NULL, there's no need for `WHERE X IS NOT NULL`, and `SELECT MAX(PK)` alone is sufficient. 

Since the NULL column is the certificate, and not the committee, this won't happen just on startup. However, I'm not sure about the frequency of this query and whether it justifies an index.

Reply:
FWIW I would keep the committee in a separate table. As I understand we are doing this without concern for performance, but there are many ways to store the committee, and some don't involve repeating the entire committee in every batch, only when it changes. To keep our options open for future changes, I would keep them separate from the beginning, rather than start messing around with existing constraints to make room for this in the same table, having to revisit the (luckily few) existing queries where we now have to add filtering.